### PR TITLE
API Frameworks

### PR DESCRIPTION
**Pull request by Staberinde**
Had an initial go at writing some content about Python API frameworks.

### CODE REVIEW

**Review by tf75 — APPROVED** [review]

File: modules/core/ApiFrameworks.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+FastAPI is a modern webserver framework, utilizing python type-hints for runtime type-checking, and with asynchronicity as a first-class use case.
+
+### Advantages of FastAPI
+
+#### Type-hints Become Single Source of Truth
+
+By using standard python type hints for runtime Request Response validation, it allows a single source of truth for the structure of the data to be represented in a human-readable way at the function level, making inconsistencies easier to pick up in advance using tools such as mypy/pyright in an IDE, and adding a justification for keeping these type hints up-to-date
+
+<!-- #### Generates OpenAPI documentation out of the box -->
```
Comment (JimmySunDance) [inline modules/core/ApiFrameworks.md:71]: Did you mean to come back and add this? Guessing you were going to talk about the swagger generation?

File: modules/core/ApiFrameworks.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+### What is FastAPI?
+
+FastAPI is a modern webserver framework, utilizing python type-hints for runtime type-checking, and with asynchronicity as a first-class use case.
+
+### Advantages of FastAPI
+
+#### Type-hints Become Single Source of Truth
+
+By using standard python type hints for runtime Request Response validation, it allows a single source of truth for the structure of the data to be represented in a human-readable way at the function level, making inconsistencies easier to pick up in advance using tools such as mypy/pyright in an IDE, and adding a justification for keeping these type hints up-to-date
+
```
Comment (JimmySunDance) [inline modules/core/ApiFrameworks.md:70]: Is it worth noting that FastAPI is one of the fastest python API frameworks? Much faster than Django and Flask.

**Review by JimmySunDance — APPROVED** [review]

File: modules/core/ApiFrameworks.md (hunk 1 above)
Comment (Staberinde) [reply modules/core/ApiFrameworks.md:71]: Sorry that shouldn't have been committed. I was debating whether it was worth adding, but given that OpenAPI/swagger instrumentation via plugin is pretty trivial in the other 2 frameworks it didn't really seem worth mentioning

File: modules/core/ApiFrameworks.md (hunk 2 above)
Comment (Staberinde) [reply modules/core/ApiFrameworks.md:70]: I'm happy to talk about performance if you think it's worth mentioning; my thinking is that it's difficult to talk about performance in very general terms as it depends to a great extent on the use case of the API

**Pull request merged**